GitCommands/UserRepositoryHistory/RemoteRepositoryManager.cs

```diff
@@ -72,16 +72,9 @@ public async Task<IList<Repository>> LoadRecentHistoryAsync()
         {
             await TaskScheduler.Default;
 
-            // BUG: this must be a separate settings
-            // TODO: to be addressed separately
-            int size = AppSettings.RecentRepositoriesHistorySize;
             IReadOnlyList<Repository> history = _repositoryStorage.Load(KeyRemoteHistory);
-            if (history is null)
-            {
-                return Array.Empty<Repository>();
-            }
-
-            return AdjustHistorySize(history, size).ToList();
+            const int remotesCacheLength = 30;
```

Comment:
I do not fully agree here, I have set dropdown to 50 so it fits one screen, same applies here.
But not blocking.

Reply:
I need to think about it again. I don't want to break someone workflow but it seems to me a setting not related so making not sense upfront...

Reply:
I can live with this too, remotes are shifted out if hardcoded to 30 or 50 that I have set

Reply:
What do you think about a hidden AppSetting instead? (I just pushed a change to do that)
That we could put in the UI later if needed....

I made this change because it feels to me very weird to use the setting of another feature for a different feature and the comment confirmed my point of view but I don't really understand this feature because, as I said, for me, the value you are interested **can't, by definition, never be available**. (Or I would like a real explanation of this feature).

Maybe it would much clever to generate a "default" value based on the existing remotes and the name of the remote just entered instead of having a huge dropdown list, don't you think?